Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+  letI mα : MeasureSpace α := ⟨μ⟩
+  letI mS : MeasureSpace (closure S) := Measure.Subtype.measureSpace
+  have : IsFiniteMeasure mS.volume := by
+    constructor
+    rw [Measure.Subtype.volume_univ (MeasurableSet.nullMeasurableSet measurableSet_closure)]
+    exact measure_lt_top volume (closure S)
+  have := of_separableSpace_on_countablyGenerated_uniformSpace (μ := mS.volume)
+  intro ε hε
+  obtain ⟨K, hK, hKe⟩ := this ε hε
+  have hSμ : μ (closure S) = μ Set.univ := le_antisymm (measure_mono <| Set.subset_univ _)
+    (le_trans hSμ.ge (measure_mono <| subset_closure))
+  have hSμ_co : μ (closure S)ᶜ = 0 := by
+    rw [MeasureTheory.measure_compl, tsub_eq_zero_of_le hSμ.ge]
+    · measurability
+    · rw [hSμ]
+      exact measure_ne_top _ _
+  use (closure K)
+  constructor
+  · apply TotallyBounded.closure
+    rw [Metric.totallyBounded_iff] at *
+    intro δ hδ
+    obtain ⟨N, hN⟩ := hK δ hδ
+    use N, Finite.image Subtype.val hN.1
+    simp_all only [iUnion_coe_set, mem_image, Subtype.exists, exists_and_right, exists_eq_right,
+      iUnion_exists, image_subset_iff, preimage_iUnion]
+    exact hN.right
+  · have hKe : volume (closure K)ᶜ ≤ ε := by
+      apply le_trans ?_ hKe
+      apply measure_mono
+      rw [Set.compl_subset_compl]
+      exact subset_closure
+    apply le_trans ?_ hKe
+    rw [Measure.Subtype.volume_def, MeasureTheory.Measure.comap_apply _ Subtype.val_injective]
+    · rw [← MeasureTheory.measure_inter_add_diff₀ (t := closure S),
+        ← MeasureTheory.measure_inter_add_diff₀ (μ := volume) (t := closure S), volume]
+      congr
+      apply add_le_add
+      · apply measure_mono
+        intro x hx
+        simp only [mem_inter_iff, mem_image, mem_compl_iff, closure_subtype, Subtype.exists,
+          exists_and_left, exists_prop, exists_eq_right_right, and_self_right]
+        constructor <;> simp_all only [mem_inter_iff, mem_compl_iff, not_false_eq_true]
+      · have : μ ((closure (Subtype.val '' K))ᶜ \ closure S) = 0 := by
+          rw [← nonpos_iff_eq_zero, ← hSμ_co]
+          apply measure_mono
+          intro x hx
+          simp_all only [mem_diff, mem_compl_iff, not_false_eq_true]
+        rw [this]
+        exact zero_le _
+      · apply MeasurableSet.nullMeasurableSet
+        measurability
+      · apply MeasurableSet.nullMeasurableSet
+        measurability
+    · exact fun s hs ↦ MeasurableSet.subtype_image measurableSet_closure hs
+    · measurability
+
+end IsPretight
+
+/-- A measure `μ` is tight if for all `0 < ε`, there exists `K` compact such that `μ Kᶜ ≤ ε`.
+This is formulated in terms of filters for simplicity, and proven equivalent to the usual definition
+in `iff_isCompact_sets`. -/
+def IsTight [TopologicalSpace α] (μ : Measure α) : Prop := Tendsto μ (cocompact α).smallSets (𝓝 0)
+
+namespace IsTight
+
+/-- The usual definition of tightness is equivalent to the filter definition. -/
+lemma iff_isCompact_sets [TopologicalSpace α] {μ : Measure α} :
```

Comment:
This name does not really correspond to the statement. `iff_exists_isCompact_measure_compl_le`? (This is maybe too long, but much easier to find).